src/AstAnalyser.js

```diff
@@ -14,10 +15,25 @@ import { JsSourceParser } from "./JsSourceParser.js";
 export class AstAnalyser {
   /**
    * @constructor
-   * @param { SourceParser } [parser]
+   * @param options
    */
-  constructor(parser = new JsSourceParser()) {
-    this.parser = parser;
+  constructor(options = {}) {
+    if (options.customParser !== undefined) {
+      assert(options.customParser instanceof JsSourceParser || typeof options.customParser === "object",
+        `customParser must be an instance of JsSourceParser or an object`);
+    }
+    if (options.customProbe !== undefined) {
+      assert(Array.isArray(options.customProbe), `customProbe must be an array`);
+    }
+    if (options.isReplacing !== undefined) {
+      assert(typeof options.isReplacing === "boolean", `isReplacing must be a boolean`);
+    }
+
+    this.parser = options.customParser || new JsSourceParser();
+    this.options = {
+      isReplacing: options.isReplacing || false,
+      customProbe: options.customProbe || []
+    };
```

Comment:
Thanks for `??` it's fixed in latest version (the assert has been removed as well to rely only on types)

i have rename the `options` property to `probeOptions` for better clarity. But maybe you were suggesting to change the structure of the options? Instead of remapping some parameters in object options while runtime between each constructor, directly consume a fix structure of options supplied initially by the user? Something like that (below) in each class ?

```js
{
  customParser: new JsSourceParser(),
  probesOptions: {
    customProbes: [],
    skipDefaultProbes: false
  },
  runtimeOptions: {
    minified: false,
    package: false,
  etc
  ...
  },
}
```

It might be link to this subject as well https://github.com/NodeSecure/js-x-ray/pull/250#discussion_r1525834884

Reply:
resolving as updated in latest commit